adapters/InMobi/InMobiAdapter/GADMAdapterInMobiUnifiedNativeAd.m

```diff
@@ -23,287 +30,301 @@ @interface GADMAdapterInMobiUnifiedNativeAd () <IMNativeDelegate>
+        return delegate;
+    };
+    
+    GADMAdapterInMobiUnifiedNativeAd *__weak weakSelf = self;
+    NSString *accountID = _nativeAdConfig.credentials.settings[GADMAdapterInMobiAccountID];
+    [GADMAdapterInMobiInitializer.sharedInstance
+     initializeWithAccountID:accountID
+     completionHandler:^(NSError *_Nullable error) {
+        GADMAdapterInMobiUnifiedNativeAd *strongSelf = weakSelf;
+        if (!strongSelf) {
+            return;
+        }
+        
+        if (error) {
+            NSLog(@"[InMobi] Initialization failed: %@", error.localizedDescription);
+            strongSelf->_nativeRenderCompletionHandler(nil, error);
+            return;
+        }
+        
+        [strongSelf requestNativeAdWithOptions:strongSelf->_nativeAdConfig.options];
+    }];
+    
 }
 
 - (void)requestNativeAdWithOptions:(nullable NSArray<GADAdLoaderOptions *> *)options {
-  id<GADMAdNetworkConnector> strongConnector = _connector;
-  id<GADMAdNetworkAdapter> strongAdapter = _adapter;
-  if (!strongConnector || !strongAdapter) {
-    return;
-  }
-
-  for (GADNativeAdImageAdLoaderOptions *imageOptions in options) {
-    if (![imageOptions isKindOfClass:[GADNativeAdImageAdLoaderOptions class]]) {
-      continue;
+    for (GADNativeAdImageAdLoaderOptions *imageOptions in options) {
+        if (![imageOptions isKindOfClass:[GADNativeAdImageAdLoaderOptions class]]) {
+            continue;
+        }
+        _shouldDownloadImages = !imageOptions.disableImageLoading;
     }
-    _shouldDownloadImages = !imageOptions.disableImageLoading;
-  }
-
-  NSString *accountID = strongConnector.credentials[GADMAdapterInMobiAccountID];
-  GADMAdapterInMobiUnifiedNativeAd *__weak weakSelf = self;
-  [GADMAdapterInMobiInitializer.sharedInstance
-      initializeWithAccountID:accountID
-            completionHandler:^(NSError *_Nullable error) {
-              GADMAdapterInMobiUnifiedNativeAd *strongSelf = weakSelf;
-              if (!strongSelf) {
-                return;
-              }
-
-              if (error) {
-                NSLog(@"[InMobi] Initialization failed: %@", error.localizedDescription);
-                [strongSelf->_connector adapter:strongAdapter didFailAd:error];
-                return;
-              }
-
-              [strongSelf requestNativeAd];
-            }];
+    
+    NSString *accountID = _nativeAdConfig.credentials.settings[GADMAdapterInMobiAccountID];
+    
+    GADMAdapterInMobiUnifiedNativeAd *__weak weakSelf = self;
+    [GADMAdapterInMobiInitializer.sharedInstance
+     initializeWithAccountID:accountID
+     completionHandler:^(NSError *_Nullable error) {
+        GADMAdapterInMobiUnifiedNativeAd *strongSelf = weakSelf;
+        if (!strongSelf) {
+            return;
+        }
+        
+        if (error) {
+            NSLog(@"[InMobi] Initialization failed: %@", error.localizedDescription);
+            strongSelf->_nativeRenderCompletionHandler(nil,error);
+            return;
+        }
```

Comment:
We already did the initialization check above, do we need to do it again?

Reply:
removed